storage/replica_command.go

```diff
@@ -546,12 +550,29 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 		// less than "f", and so the meta row {"f"->["c", "f")} would be
 		// ignored by MVCCReverseScan. In the second case, the range descriptor
 		// received by MVCCScan will be filtered before results are returned.
+		// Example: ranges ["c", "f") and ["f", "z"), reverse scan on ["d", "f"]
+		// leads to inspecting the descriptor {"z"->["f","z")}, which is
+		// discarded since it's not being asked for.
 		startKey, endKey := keys.MetaScanBounds(args.Key)
 		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, 1,
 			args.Timestamp, consistent, args.Txn)
 		if err != nil {
 			return reply, nil, err
 		}
+		var firstRD proto.RangeDescriptor
+		if err = gogoproto.Unmarshal(kvs[0].Value.Bytes, &firstRD); err != nil {
+			return reply, nil, err
+		}
+		if keys.RangeMetaKey(firstRD.StartKey).Less(args.Key) {
```

Comment:
Ah, I see. Thanks for the explanation! Sure, will update the comment. 

Change to `RangeLookup` looks good to me!